### PR TITLE
Custom component can receive onAccept prop using a children function

### PR DESCRIPTION
**Pull request by lvauvillier**
There was no possibility to bind the onAccept callback to a custom element to dismiss the banner.

This PR solves this allowing to set a function as children with a onAccept argument.
This function must return a valid react element.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "react/no-unknown-property": 1,
     "react/react-in-jsx-scope": 1,
     "react/self-closing-comp": 1,
-    "react/wrap-multilines": 1
+    "react/wrap-multilines": 1,
+    "react/no-multi-comp": 0
```
Comment (gabro) [inline .eslintrc]: why is this necessary?

File: test/tests/CookieBanner-test.js [hunk 2]
```diff
@@ -111,4 +111,33 @@ describe('CookieBanner', () => {
...
+    });
+
+    MyOtherComponent.propTypes = {
+      onAccept: PropTypes.func
+    };
+
+    const component = (
+      <div>
+        <CookieBanner>
+          {(onAccept) => { return (<MyOtherComponent onAccept={onAccept} />); }}
```
Comment (gabro) [inline test/tests/CookieBanner-test.js]: this can use a shorter syntax

```{onAccept => <MyOtherComponent onAccept={onAccept} />}```

File: src/CookieBanner.js [hunk 3]
```diff
@@ -169,6 +172,9 @@ export default class CookieBanner extends React.Component {
   getBanner = () => {
     const { children, className, message } = this.props;
     if (children) {
+      if (typeof children === 'function') {
```
Comment (gabro) [inline src/CookieBanner.js]: use `t.Function.is(children)`

File: test/tests/CookieBanner-test.js [hunk 4]
```diff
@@ -111,4 +111,33 @@ describe('CookieBanner', () => {
     expect(_myComponent.length).toBe(1, 'cookie banner is not displaing custom child component');
   });
 
+  it('should be replaced with custom child component using function', () => {
+
+    const MyOtherComponent = React.createClass({
```
Comment (gabro) [inline test/tests/CookieBanner-test.js]: For consistency with the rest of the cose base, please use ES6 classes for components.

In this case you can actually use a stateless component:

```js
const MyOtherComponent = ({ onAccept }) => <div className='my-other-component' onClick={this.props.onAccept} />;
```

File: test/tests/CookieBanner-test.js [hunk 5]
```diff
@@ -111,4 +111,33 @@ describe('CookieBanner', () => {
...
+      render() {
+        return <div className='my-other-component' onClick={this.props.onAccept}/>;
+      }
+    });
+
+    MyOtherComponent.propTypes = {
```
Comment (gabro) [inline test/tests/CookieBanner-test.js]: `PropTypes` don't add much value to the test here. I would just remove this part to keep the code shorter.

(in general - by the way - we use @props from `tcomb-react` instead of `PropTypes`)

File: test/tests/CookieBanner-test.js [hunk 6]
```diff
@@ -111,4 +111,33 @@ describe('CookieBanner', () => {
...
+      </div>
+    );
+
+    const cookieWrapper = TestUtils.renderIntoDocument(component);
+
+    const banner = TestUtils.scryRenderedDOMComponentsWithClass(cookieWrapper, 'react-cookie-banner');
+    expect(banner.length).toBe(0, 'cookie banner is being displayed');
+
+    const _myComponent = TestUtils.scryRenderedDOMComponentsWithClass(cookieWrapper, 'my-other-component');
+    expect(_myComponent.length).toBe(1, 'cookie banner is not displaing custom child component using function');
```
Comment (gabro) [inline test/tests/CookieBanner-test.js]: typo: displaing -> displaying

File: test/tests/CookieBanner-test.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import React from 'react/addons';
+import React, { PropTypes } from 'react/addons';
```
Comment (gabro) [inline test/tests/CookieBanner-test.js]: see comment below, we probably don't need `PropTypes`

File: test/tests/CookieBanner-test.js (hunk 4 above)
Comment (lvauvillier) [inline test/tests/CookieBanner-test.js]: you depends on react 0.13 that does not allow stateless components.

File: .eslintrc (hunk 1 above)
Comment (lvauvillier) [inline .eslintrc]: eslint doesn't allow more than one react component definition per file. MyOtherComponent definition raise an error.

File: test/tests/CookieBanner-test.js (hunk 6 above)
Comment (lvauvillier) [inline test/tests/CookieBanner-test.js]: done

File: src/CookieBanner.js (hunk 3 above)
Comment (lvauvillier) [inline src/CookieBanner.js]: done

File: test/tests/CookieBanner-test.js (hunk 5 above)
Comment (lvauvillier) [inline test/tests/CookieBanner-test.js]: done

File: test/tests/CookieBanner-test.js (hunk 7 above)
Comment (lvauvillier) [inline test/tests/CookieBanner-test.js]: done

**Comment by lvauvillier** [comment]
Thanks for your review. I just push a new version.
Eslint now raise an error due to the definition of MyOtherComponent.
Tell me if you prefer adding a new rule, or a eslint-disable-line comment to catch this error.

File: test/tests/CookieBanner-test.js (hunk 2 above)
Comment (lvauvillier) [inline test/tests/CookieBanner-test.js]: done

File: .eslintrc (hunk 1 above)
Comment (gabro) [inline .eslintrc]: ok, it's clearer now. I'll take care of this, thanks!

File: test/tests/CookieBanner-test.js [hunk 8]
```diff
@@ -108,7 +108,32 @@ describe('CookieBanner', () => {
     expect(banner.length).toBe(0, 'cookie banner is being displayed');
 
     const _myComponent = TestUtils.scryRenderedDOMComponentsWithClass(cookieWrapper, 'my-component');
-    expect(_myComponent.length).toBe(1, 'cookie banner is not displaing custom child component');
+    expect(_myComponent.length).toBe(1, 'cookie banner is not displaying custom child component');
+  });
+
+  it('should be replaced with custom child component using function', () => {
+
+    const MyOtherComponent = React.createClass({
```
Comment (gabro) [inline test/tests/CookieBanner-test.js]: please make this an ES6 class, for consistency with the rest of the code-base

File: test/tests/CookieBanner-test.js (hunk 4 above)
Comment (gabro) [inline test/tests/CookieBanner-test.js]: right, we should probably make this an ES6 class then.

**Comment by gabro** [comment]
@lvauvillier I pushed a commit to your branch that should fix the ESLint issue ;)

File: test/tests/CookieBanner-test.js (hunk 8 above)
Comment (gabro) [inline test/tests/CookieBanner-test.js]: Actually, sorry, I just realized this is not your fault but you just copy-pasted the example above. This is our mistake, and I'll fix it in a subsequent PR.!

**Comment by gabro** [comment]
Thanks @lvauvillier 🎉
